### PR TITLE
Check runtime ancestors used in annotations files

### PR DESCRIPTION
**Pull request by Morriar**
### Type of Change

- [ ] Add RBI for a new gem
- [ ] Modify RBI for an existing gem
- [x] Other: Ci

### Changes

When running runtime checks we also check that super classes and modules included/extended in the RBI file reflect the runtime behavior.

This allowed me to find a few errors in the current set of annotations (see discussions below).

Closes https://github.com/Shopify/rbi-central/issues/81.

### CODE REVIEW

File: rbi/annotations/lhm-shopify.rbi [hunk 1]
```diff
@@ -69,7 +69,7 @@ class Lhm::Printer::Base
   def initialize; end
 end
 
-class Lhm::Printer::Percentage < Lhm::Printer::Base
+class Lhm::Printer::Percentage
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:72]: See https://github.com/Shopify/lhm/blob/master/lib/lhm/printer.rb#L15

File: rbi/annotations/parse-cron.rbi [hunk 2]
```diff
@@ -71,8 +69,6 @@ end
 
 # internal "mutable" time representation
 class CronParser::InternalTime
-  extend T::Generic
```
Comment (Morriar) [inline rbi/annotations/parse-cron.rbi:74]: See https://github.com/siebertm/parse-cron/blob/master/lib/cron_parser.rb (not generic)

File: rbi/annotations/parse-cron.rbi [hunk 3]
```diff
@@ -2,8 +2,6 @@
 
 # Parses cron expressions and computes the next occurence of the "job"
 class CronParser
-  extend T::Generic
```
Comment (Morriar) [inline rbi/annotations/parse-cron.rbi:5]: See https://github.com/siebertm/parse-cron/blob/master/lib/cron_parser.rb (not generic)

File: rbi/annotations/actionview.rbi [hunk 4]
```diff
@@ -3,7 +3,7 @@
 module ActionView
   TemplateError = T.type_alias { Template::Error }
 
-  class MissingTemplate < ActionViewError
+  class MissingTemplate < ActionView::ActionViewError
```
Comment (Morriar) [inline rbi/annotations/actionview.rbi:6]: The problem here is that the class is not flattened properly and thus the superclass name is not fully qualified. We could be smarter but I think it's a good thing to enforce fully qualified names

File: rbi/annotations/rainbow.rbi [hunk 5]
```diff
@@ -15,7 +15,7 @@ module Rainbow
     sig { params(hex: String).returns([Integer, Integer, Integer]) }
     def self.parse_hex_color(hex); end
 
-    class Indexed < Color
+    class Indexed < Rainbow::Color
```
Comment (Morriar) [inline rbi/annotations/rainbow.rbi:18]: Same here, the class is not flattened properly so the names are not fully qualified.

File: index.json [hunk 6]
```diff
@@ -109,7 +109,9 @@
       "minitest"
     ],
     "requires": [
-      "minitest/test"
+      "minitest/test",
```
Comment (Morriar) [inline index.json:112]: Those requires where required to be able to load the constants used in the annotations file.

**Review by KaanOzkan — APPROVED** [review]
<3

**Review by vinistock — APPROVED** [review]

**Pull request merged**